src/metadata.rs

```diff
@@ -1,81 +1,431 @@
+    pub fn get_possible_metric_details(&self, metrics: &[MetricId]) -> Result<DataFrame> {
+        let mut id_collections: HashMap<String, Vec<String>> = HashMap::new();
+
+        for metric in metrics {
+            id_collections
+                .entry(metric.to_col_name())
+                .and_modify(|e| e.push(metric.to_query_string().into()))
+                .or_insert(vec![metric.to_query_string().into()]);
+        }
+
+        let mut filter_expression: Option<Expr> = None;
+        for (col_name, ids) in &id_collections {
+            let filter_series = Series::new("filter", ids.clone());
+            filter_expression = if let Some(expression) = filter_expression {
+                Some(expression.or(col(col_name).is_in(lit(filter_series))))
+            } else {
+                Some(col(col_name).is_in(lit(filter_series)))
+            };
+        }
+
+        let metrics = self
+            .combined_metric_source_geometry()
+            .filter(filter_expression.unwrap())
+            .collect()?;
+        Ok(metrics)
+    }
+
+    /// Given a geometry level return the path to the
+    /// geometry file that it corrisponds to
+    pub fn get_geom_details(&self, geom_level: &str) -> Result<String> {
+        let matches = self
+            .geometries
+            .clone()
+            .lazy()
+            .filter(col("level").eq(lit(geom_level)))
+            .collect()?;
+
+        let file = matches
+            .column("filename_stem")?
+            .str()?
+            .get(0)
+            .unwrap()
+            .into();
+
+        Ok(file)
+    }
+}
+
+impl CountryMetadataLoader {
+    /// Create a metadata loader for a specific Country
+    pub fn new(country: &str) -> Self {
+        let paths = CountryMetadataPaths::default();
+        Self {
+            country: country.into(),
+            paths,
+        }
+    }
+    /// Overwrite the Paths object to specifiy custom
+    /// metadata filenames and `base_url`.
+    pub fn with_paths(&mut self, paths: CountryMetadataPaths) -> &mut Self {
+        self.paths = paths;
+        self
+    }
+
+    /// Load the Metadata catalouge for this country with
+    /// the specified metadata paths
+    pub fn load(&self) -> Result<Metadata> {
+        Ok(Metadata {
+            metrics: self.load_metadata(&self.paths.metrics)?,
+            geometries: self.load_metadata(&self.paths.geometry)?,
+            source_data_releases: self.load_metadata(&self.paths.source_data)?,
+            data_publishers: self.load_metadata(&self.paths.data_publishers)?,
+            countries: self.load_metadata(&self.paths.country)?,
+        })
+    }
+
+    /// Performs a load of a given metadata parquet file
+    fn load_metadata(&self, path: &str) -> Result<DataFrame> {
+        let url = format!("{}/{}/{path}", self.paths.base_url, self.country);
+        let args = ScanArgsParquet::default();
+        println!("Attempting to load {url}");
```

Comment:
Update with logging:
```suggestion
        info!("Attempting to load {url}");
```

Reply:
Logging is in the #38 PR now. I think we should merge this in and #38 then do a sweep to update any println! to appropriate logging levels